### DISCUSSION TITLE
Add a `Path` property to store shape's info

### DISCUSSION THREAD

**Discussion by BiDuang** [ShapeCrawler/ShapeCrawler#589] [Ideas]
How about add a `Path` property to store shape's `SlidePart` uri and `Xpath`. (Just like `DocumentFormat.OpenXml.XmlPath`)

It can be a unique index to operate on an element or element navigation. I'd love to try to open a PR.

**Comment by AdamShakhabov** [comment]
I would suggest adding two separate properties for this, such as `IShape.SDKSlidePart` and `IShape.SDKXPath`.

I hope I have understood your idea correctly.

And yes, PRs are welcome 👍

**Reply by BiDuang** [reply]
Yep, I totally agree that. But I found that `ITextFrame` doesn't inherit from any `IShape` class (Actually it inherits nothing), so if I call `GetAllTextFrame()` and then try to get its path info, the `IShape.SDKSlidePart` and `IShape.SDKXPath` won't work.

Do you have any good solution?

**Reply by AdamShakhabov** [reply]
Then what about creating a new `IPath` interface with the property `IPath.SDKXPath` and inherit `IShape` and `ITextFrame` interfaces from `IPath`?

**Reply by BiDuang** [reply]
I found that `IShape` has already implement a property called `SlideStructure`, so `IShape.SDKSlidePart` may be unnecessary?  

How about this:

```c#
namespace ShapeCrawler.Shapes;

/// <summary>
///     Represents path information of the shape.
/// </summary>
public interface IPath
{
    /// <summary>
    ///     Gets the shape's XPath in the slide.
    /// </summary>
    public string SDKXPath { get; }
}
```

**Reply by AdamShakhabov** [reply]
I'm making a big refactoring (https://github.com/ShapeCrawler/ShapeCrawler/pull/561) in parallel. The property `ISlide.SlideStructure` will be deleted.

**Reply by BiDuang** [reply]
Oh, then how to get the shape's parent slide infomation after the refactoring merged?

**Reply by AdamShakhabov** [reply]
Add also `IShape.SDKSlidePart`. I will merge this changes in my refactoring and will keep this property. 

The refactoring is not completed yet.

**Reply by BiDuang** [reply]
So I just push a PR with `IPath.SDKXPath` and leave the `IShape.SDKSlidePart` part until you're done refactoring? Or push PR directly on branch `540-add-slide-number`?

**Reply by AdamShakhabov** [reply]
Do not wait merging of `540-add-slide-number`. You can push now.
